### ISSUE TITLE
A potential risk in loxilb  that could lead to takeover of the cluster 

### ISSUE THREAD

**Issue by HouqiyuA** [loxilb-io/loxilb#852]
Dear Team Members:

Greetings! Our team is very interested in your project and we recently identified a potential RBAC security risk while doing a security assessment of your project. Therefore, we would like to report it to you and provide you with the relevant details so that you can fix and improve it accordingly.

Details:

In this Kubernetes project, there is a ClusterRole named “loxilb-ingress” that has high-risk permissions to list, acquire, and monitor secret resources. (https://github.com/loxilb-io/loxilb/blob/main/cicd/k3s-flannel-loxilb-ingress/ingress/loxilb-ingress-deploy.yml. This ClusterRole has direct permission to list entire cluster secrets. There is also a ClusterRole named “multus” (https://github.com/loxilb-io/loxilb/blob/main/cicd/k3s-sctpmh-seagull/). multus/multus-daemonset.yml) that has arbitrary permissions to any resource.
These authorizations are high-risk, and if an attacker steals a service account with these permissions, he can elevate the privileges and further take over the entire cluster.

Please confirm the purpose of the assignment of this permission and consider using a more granular permission non-assignment rule.
Best wishes.

HouqiyuA

**Comment by backguynn** [comment]
Thank you for your valuable feedback. 

Based on your suggestion, we have removed some unnecessary resource access permissions from loxilb-ingress. (https://raw.githubusercontent.com/loxilb-io/loxilb-ingress/refs/heads/main/manifests/loxilb-ingress-deploy.yml)  

As for the ClusterRole of Multus, we have used the manifest file provided by the Multus project without modification. This file is not related to the Loxilb project. 

Additionally, any discretionary permissions requested by Multus apply only to resources in the k8s.cni.cncf.io API group. This API group pertains to the NetworkAttachmentDefinition CRD. NetworkAttachmentDefinition is a CRD schema specified by the Network Plumbing Working Group to express the intent to connect a Pod to one or more logical or physical networks.
